doc/manual/chapters/translation.tex

```diff
@@ -151,58 +513,44 @@ \subsubsection{Sets}
 \noindent
 \begin{minipage}[t]{\textwidth}
 \begin{lstlisting}[language=golang]
-import "pgo/datatypes"
-
-var S datatypes.Set = datatypes.NewSet(1, 5, 6)
-var T datatypes.Set = datatypes.NewSet(2, 3)
-var U datatypes.Set = datatypes.SetConstructor(func(x int) bool {
-		return x > 3
-	}, S)
-var V datatypes.Set = datatypes.SetImage(func(x int, y int) int {
-		return x + y
-	}, S, T)
+S := []int{1, 5, 6}
+T := []int{2, 3}
+tmpSet := make([]int, 0)
+for _, x := range S {
+	if x > 3 {
+		tmpSet = append(tmpSet, x)
+	}
+}
+U := tmpSet
+tmpSet0 := make([]int, 0)
+for _, x := range S {
+	for _, y := range T {
+		tmpSet0 = append(tmpSet0, x+y)
+	}
+}
+sort.Ints(tmpSet0)
+if len(tmpSet0) > 1 {
+	previousValue := tmpSet0[0]
+	currentIndex := 1
+	for _, v := range tmpSet0[1:] {
+		if previousValue != v {
+			tmpSet0[currentIndex] = v
+			currentIndex++
+		}
+		previousValue = v
+	}
+	tmpSet0 = tmpSet0[:currentIndex]
+}
+V := tmpSet0
 // ...
 \end{lstlisting}
 \captionof{figure}{Compiled Go}
 \end{minipage}
 
 While not as concise as the PlusCal, the output Go code is still readable.
 
-The Cartesian product operator \verb|\X| in PlusCal is not associative. For example,
-\begin{itemize}
-\item \texttt{<<1, 2, 3>>} is in \verb|Nat \X Nat \X Nat|,
-
-\item \texttt{<< <<1, 2>>, 3>>} is in \verb|(Nat \X Nat) \X Nat|, and
-
-\item \texttt{<<1, <<2, 3>> >>} is in \verb|Nat \X (Nat \X Nat)|,
-\end{itemize}
-and none of these are equivalent. In Go, \texttt{S.CartesianProduct(T, U, V, ...)} is equivalent to \verb|S \X T \X U \X ...|. PGo composes these function calls to create the parenthesized expressions above.
-
-\noindent
-\begin{minipage}[t]{\textwidth}
-\begin{lstlisting}[language=pcal]
-variables
-	S = {1, 2, 3};
-	T = S \X S \X S; \* << 1, 3, 2 >> \in T
-	U = S \X (S \X S); \* << 1, << 3, 2 >> >> \in U
-\end{lstlisting}
-\captionof{figure}{PlusCal}
-\end{minipage}
-
-\noindent
-\begin{minipage}[t]{\textwidth}
-\begin{lstlisting}[language=golang]
-import "pgo/datatypes"
-
-var S datatypes.Set = datatypes.NewSet(1, 2, 3)
-var T datatypes.Set = S.CartesianProduct(S, S)
-var U datatypes.Set = S.CartesianProduct(S.CartesianProduct(S))
-\end{lstlisting}
-\captionof{figure}{Compiled Go}
-\end{minipage}
-
-\subsubsection{Maps}
-PlusCal ``functions'' are just maps in Go. \texttt{pgo/datatypes} implements an ordered, thread-safe map in Go, which supports using slices and tuples as keys (unlike the builtin Go map). There are several syntaxes to declare maps, which are handled by library methods similar to the set methods.
+\subsubsection{Functions}
+PlusCal functions with finite domains are translated into sorted slices of structs in Go.
```

Comment:
I would not call these "PlusCal functions", since the text refers to TLA+ functions, and we have talked about them as such in the "Overview" table (same for the other data types).